FS.ios.js

```diff
@@ -23,48 +16,138 @@ var convertError = (err) => {
 var NSFileTypeRegular = RNFSManager.NSFileTypeRegular;
 var NSFileTypeDirectory = RNFSManager.NSFileTypeDirectory;
 
+/**
+ * This function converts the relative path to application home specific directory
+ * Ex: Documents/hello.txt -> <DocumentsDirForApp>/hello.txt
+ * Caches/hello.txt -> <CacheDirForApp>/hello.txt
+ * tmp/hello.txt -> <TmpDirForApp>/hello.txt
+ * @param  {String}
+ * @return {String}
+ */
+function processFilePath(filepath) {
+  var found = filepath.indexOf('/');
+  if (found !== 0) {
+    //relative path, check where is it relative to
```

Comment:
Using a `switch` statement here would clean thing up a bit:

```javascript
var splitPath = filepath.split('/');
switch (splitPath[0]) {
  case 'Documents':
    splitPath[0] = RNFSManager.NSDocumentDirectoryPath;
    break;
  case 'Caches':
    splitPath[0] = RNFSManager.NSCachesDirectoryPath;
    break;
  case 'tmp':
    splitPath[0] = RNFSManager.NSTmpDirectoryPath;
    break;
  default:
    return RNFSManager.NSDocumentDirectoryPath + '/' + filepath;
}

return splitPath.join('/');
```

Seems much more readable to me.